### DISCUSSION TITLE
Format of binary data to ubxfile.py

### DISCUSSION THREAD

**Discussion by n8ur** [semuconsulting/pyubx2#103] [Q&A]
I've been given a large file that is a hex dump formatted in lines of 24 two-byte space-separated hex values with a newline at the end of each line.  An example of the line formating is: "b2 49 b6 42 49 b6 49"

The data is a stream captured from the SDA line of an I2C port, and it looks like there is a long and variable length sequence of 00 or ff bytes between actual messages (I think just capturing whatever's on the bus at each clock tick).

I've tried munging this a couple of different ways like removing all spaces and newlines, or preceding each byte with '\x', but any of them yield 0 messages read when run through ubxfile.py.

How should a file like this be formatted for the pyubx2 tools to parse it?

Thanks!
John

**Comment by semuadmin** [comment]
Hi @n8ur

Unfortunately you may be out of luck here - pyubx2 isn't designed to handle raw I2C data streams. You'd need some kind of I2C-UART protocol adapter to produce a stream of raw UBX data b"\xb5\x62..." etc.

FYI I did have a play with the [AdaFruit multi-protocol adapter](https://www.adafruit.com/product/2264) to see if it might provide a suitable capability, but it requires a relatively complex set-up using the libusb library so wasn't really a turnkey solution. You may have more luck with it.
